arcgisUM.py

```diff
@@ -0,0 +1,206 @@
+import traceback
+
+def handleError(self, record):  # @UnusedVariable
+    traceback.print_stack()
+    
+logging.Handler.handleError = handleError
+#####
+
+TIMEZONE_UTC = dateutil.tz.tzutc()
+RUN_START_TIME = datetime.datetime.now(tz=TIMEZONE_UTC)
+RUN_START_TIME_FORMATTED = RUN_START_TIME.strftime('%Y%m%d%H%M%S')
+
+# Hold parsed options
+options = None
+
+# TODO: required in this module?
+courseLogHandlers = dict()
+courseLoggers = dict()
+
+def getArcGISConnection(securityinfo):
+    """
+    Get a connection object for ArcGIS based on configuration options
+    
+    :return: Connection object for the ArcGIS service
+    :rtype: arcresthelper.orgtools.orgtools
+    :raises: RuntimeError if ArcGIS connection is not valid
+    :raises: arcresthelper.common.ArcRestHelperError for ArcREST-related errors
+    """
+
+    if not isinstance(securityinfo, dict):
+        raise TypeError('Argument securityinfo type should be dict')
+
+    try:
+        arcGIS = GIS(securityinfo['org_url'],
+                     securityinfo['username'],
+                     securityinfo['password']);
+    except RuntimeError as exp:
+        logger.error("RuntimeError: getArcGISConnection: {}".format(exp))
+        raise RuntimeError(str('ArcGIS connection invalid: {}'.format(exp)))
+    
+    return arcGIS
+
+def getArcGISGroupByTitle(arcGISAdmin, title):
+    """
+    Given a possible title of a group, search for it in ArcGIS
+    and return a Group object if found or None otherwise.
+
+    :param arcGISAdmin: ArcGIS Administration REST service connection object
+    :type arcGISAdmin: arcrest.manageorg.administration.Administration
+    :param title: Group title to be found
+    :type title: str
+    :return: ArcGIS Group object or None
+    :rtype: Group or None
+    """
+    searchString = "title:"+title
+    logger.debug("group search string: original: {}".format(searchString))
+
+    # quote characters that are special in the group search.
+    searchString = searchString.translate(str.maketrans({"?":  r"\?","*":  r"\*"}))
+
+    logger.debug("group search string: escaped: {}".format(searchString))
+    
+    try:
+        gis_groups = arcGISAdmin.groups.search(searchString)
+    except RuntimeError as exp:
+        logger.error("arcGIS error finding group: {} exception: {}".format(searchString,exp))
+    
+    if len(gis_groups) > 0:
+        return gis_groups.pop()
+
+    return None
+
+
+def addCanvasUsersToGroup(instructorLog, group, courseUsers):
+    """Add new users to the ArcGIS group.  """
+    groupNameAndID = util.formatNameAndID(group)
+    
+    logger.info("addCanvasUsersToGroup: enter")
+    
+    if len(courseUsers) == 0:
+        logger.info('No new users to add to ArcGIS Group {}'.format(groupNameAndID))
+        return instructorLog
+
+    logger.info('Adding Canvas Users to ArcGIS Group {}: {}'.format(groupNameAndID, courseUsers))
+    # ArcGIS usernames are U-M uniqnames with the ArcGIS organization name appended.
+    user="NULL"
```

Comment:
What is the reason for the `NULL` string and not `None`

Reply:
This is unused and will be deleted.